sdk/python/tests/integration/feature_repos/repo_configuration.py

```diff
@@ -245,9 +245,9 @@ def get_local_server_port(self) -> int:
 
 def table_name_from_data_source(ds: DataSource) -> Optional[str]:
     if hasattr(ds, "table_ref"):
-        return ds.table_ref
+        return ds.table_ref  # type: ignore
```

Comment:
Nitpick, but why are we adding an ignore statement here? Was this an omission? We've not had to use ignores elsewhere in the code base, so this stood out.

Reply:
This was because I believe neither `table` not `table_ref` are actually attributes of `DataSource` - only its subclasses. The other option would have been a complex `cast`, I believe. This might be preferable but especially as it was in a test util I decided not to do that.

Reply:
There area handful of other `# type: ignore` comments around the codebase and I've also discovered that the typing around Protobufs is tricky and often seemingly wrong or unhelpful.